File: ООП/Homework2/all_animals/cat.java
package all_animals;

public  class cat extends pets{

    private final String coat_color;

    protected cat(String name, String breed, String vaccination, String Bday, int height, int weight, String eyes_color, String coat_color){
        super(name, breed, vaccination, Bday, height, weight, coat_color);
        this.coat_color = coat_color;
    }

    public void petme() {
        System.out.println("\nПогладь кошечку =^_^=");
    }

    public void voice(){
        System.out.println("Мяу");
    }

    public String toString(){
        return String.format("%s\nЦвет шерсти: %s", super.toString(), this.coat_color);
    }
}

File: ООП/Homework2/animals_PY/classes/pets/cat.py
from .pet import pet

class cat(pet):
    def __init__(self, name, height, weight, coloreye, nickname, breed, vaccination, wool, bday):
        super().__init__(name, height, weight, coloreye, nickname, breed, vaccination, wool, bday)

    def voice(self):
       print(f"{super().voice()} Мяу")

    def pet_me(self):
        return super().pet_me()
    
    def __str__(self) -> str:
        return super().__str__()